symplyphysics/laws/thermodynamics/free_energy_differential.py

```diff
@@ -46,7 +51,57 @@
     -1 * entropy * temperature_change - pressure * volume_change + chemical_potential * particle_count_change,
 )
 
-# TODO: derive from the definition of Helmholtz free energy and internal energy differential
+# Derive from the definition of Helmholtz free energy and internal energy differential
+
+_internal_energy_sym = symbols("internal_energy", cls=SymFunction)
+_entropy_change = SymSymbol("entropy_change")
+_temperature = SymSymbol("temperature")
+_volume = SymSymbol("volume")
+_particle_count = SymSymbol("particle_count")
+
+_internal_energy = _internal_energy_sym(entropy, _volume, _particle_count)
+
+_free_energy = free_energy_def.law.rhs.subs({
+    free_energy_def.internal_energy: _internal_energy,
+    free_energy_def.temperature: _temperature,
+    free_energy_def.entropy: entropy,
+})
+
+_free_energy_change = (
+    _free_energy.diff(_temperature) * temperature_change
+    + _free_energy.diff(entropy) * _entropy_change
+    + _free_energy.diff(_volume) * volume_change
+    + _free_energy.diff(_particle_count) * particle_count_change
+)
+
+_internal_energy_diff = internal_energy_differential.law.rhs.subs({
+    internal_energy_differential.temperature: _temperature,
+    internal_energy_differential.entropy_change: _entropy_change,
+    internal_energy_differential.pressure: pressure,
+    internal_energy_differential.volume_change: volume_change,
+    internal_energy_differential.chemical_potential: chemical_potential,
+    internal_energy_differential.particle_count_change: particle_count_change,
+})
+
+_internal_energy_diff_entropy = _internal_energy_diff.subs({
+    _entropy_change: 1, volume_change: 0, particle_count_change: 0
+})
+
+_internal_energy_diff_volume = _internal_energy_diff.subs({
+    _entropy_change: 0, volume_change: 1, particle_count_change: 0
+})
+
+_internal_energy_diff_particle_count = _internal_energy_diff.subs({
+    _entropy_change: 0, volume_change: 0, particle_count_change: 1
+})
+
+_free_energy_change = _free_energy_change.subs({
+    _internal_energy.diff(entropy): _internal_energy_diff_entropy,
```

Comment:
I think it is better to use `Derivative(_internal_energy(entropy), entropy)` instead of `diff()` if you are not really interested in the result.

Reply:
`_internal_energy` is a symbol so there is no differentiation happening anyway but it looks less cluttered